core/src/main/java/org/kiwix/kiwixmobile/core/search/SearchFragment.kt

```diff
@@ -104,13 +113,71 @@ class SearchFragment : BaseFragment() {
       adapter = searchAdapter
       layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
       setHasFixedSize(true)
+      // Add scroll listener to detect when the last item is reached
+      addOnScrollListener(object : RecyclerView.OnScrollListener() {
+        override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+          super.onScrolled(recyclerView, dx, dy)
+
+          val layoutManager = recyclerView.layoutManager as LinearLayoutManager
+          val totalItemCount = layoutManager.itemCount
+          val lastVisibleItem = layoutManager.findLastVisibleItemPosition()
+          // Check if the user is about to reach the last item
+          if (!isDataLoading && totalItemCount <= lastVisibleItem + VISIBLE_ITEMS_THRESHOLD) {
+            // Load more data when the last item is almost visible
+            loadMoreSearchResult()
```

Comment:
Just an idea, currently we are loading the search results if user is in the last items. What if start the loading before the 3 last items. 

```
 if (!isDataLoading && totalItemCount  <= lastVisibleItem + VISIBLE_ITEMS_THRESHOLD - 3) {
```
What do you think @MohitMaliDeveloper ?

Reply:
I had this in mind but I forgot to implement this my bad.

Reply:
Done